SourceCode/GPS/Classes/CABCurve.cs

```diff
@@ -188,26 +192,40 @@ public async void BuildCurveCurrentList(vec3 pivot)
 
                     distAway += (0.5 * widthMinusOverlap);
 
-                    curList = await Task.Run(() => BuildNewOffsetList(distAway, track));
-                    isBusyWorking = false;
-                    findGlobalNearestCurvePoint = true;
+                    if (cts != null)
+                    {
+                        cts.Cancel();
+                    }
+                    cts = new CancellationTokenSource();
+
+                    if (build != null) await build;
+
+                    build = Task.Run(() => BuildNewOffsetList(distAway, track, cts), cts.Token);
+                    curList = await build;
 
                     if (mf.isSideGuideLines && mf.camera.camSetDistance > mf.tool.width * -400)
                     {
+                        if (buildList != null)
+                            await buildList;
                         //build the list list of guide lines
-                        guideArr = await Task.Run(() => BuildCurveGuidelines(distAway, mf.ABLine.numGuideLines, track));
+                        buildList = Task.Run(() => BuildCurveGuidelines(distAway, mf.ABLine.numGuideLines, track, cts), cts.Token);
+                        guideArr = await buildList;
                     }
                     else
                     {
+                        if (buildList != null) await buildList;
                         guideArr?.Clear();
                     }
 
                 }
             }
         }
 
-        public List<vec3> BuildNewOffsetList(double distAway, CTrk track)
+        
+        public List<vec3> BuildNewOffsetList(double distAway, CTrk track, CancellationTokenSource cts = null)
```

Comment:
Methods that can be cancelled usually take a `CancellationToken` and not the `CancellationTokenSource` (which is used outside the method to actually cancel the execution).
```suggestion
        public List<vec3> BuildNewOffsetList(double distAway, CTrk track, CancellationToken ct = CancellationToken.None)
```